python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -1655,12 +1640,33 @@ def run(self, ccdExposure, *, dnlLUT=None, bias=None, deferredChargeCalib=None,
             isrFunctions.widenSaturationTrails(ccdExposure.getMaskedImage().getMask())
 
         # Brighter/Fatter
-        # FIXME: watch out (again) for gain units.
         # Units: electron
         if self.config.doBrighterFatter:
-            self.log.info("Applying Bright-Fatter kernels.")
+            self.log.info("Applying brighter-fatter correction.")
+
             bfKernelOut, bfGains = self.getBrighterFatterKernel(detector, bfKernel)
-            ccdExposure = self.applyBrighterFatterCorrection(ccdExposure, flat, dark, bfKernelOut, bfGains)
+
+            # Needs to be done in electrons; applyBrighterFatterCorrection
+            # will convert the image if necessary.
+            if exposureMetadata["LSST ISR UNITS"] == "electron":
+                brighterFatterApplyGain = False
+            else:
+                brighterFatterApplyGain = True
+                exposureMetadata["LSST ISR UNITS"] == "electron"
+
+            if brighterFatterApplyGain and (ptc is not None) and (not bfGains == gains):
```

Comment:
I don't love `not bfGains == gains`.  I guess these are dictionaries?  So this will work, but `bfGains != gains` is clearer.  And I think you should comment below that these are amp-keyed dictionaries of gains.